tests/ci/integration/tpm2_tss_patch/aws-lc-tpm2-tss.patch

```diff
@@ -29,98 +29,65 @@ index 6cc3efc8..9029383d 100644
+     X509_STORE_CTX        *ctx = NULL;
+     X509_CRL              *crl_intermed = NULL;
+     X509_CRL              *crl_ek = NULL;
+-    int                    i;
++    size_t                    i;
+     size_t                 ui;
      AUTHORITY_INFO_ACCESS *info = NULL;
-     ASN1_IA5STRING *uri = NULL;
+     ASN1_IA5STRING        *uri = NULL;
 diff --git a/src/tss2-fapi/ifapi_verify_cert_chain.c b/src/tss2-fapi/ifapi_verify_cert_chain.c
-index b495e512..c741c8ad 100644
+index df295d99..3fb475dc 100644
 --- a/src/tss2-fapi/ifapi_verify_cert_chain.c
 +++ b/src/tss2-fapi/ifapi_verify_cert_chain.c
-@@ -64,7 +64,7 @@ char* get_issuer_url(X509 *cert) {
-     AUTHORITY_INFO_ACCESS *info = X509_get_ext_d2i(cert, NID_info_access, NULL, NULL);
-     if (!info) return NULL;
+@@ -65,7 +65,7 @@ get_issuer_url(X509 *cert) {
+     if (!info)
+         return NULL;
  
 -    for (int i = 0; i < sk_ACCESS_DESCRIPTION_num(info); i++) {
 +    for (size_t i = 0; i < sk_ACCESS_DESCRIPTION_num(info); i++) {
          ACCESS_DESCRIPTION *ad = sk_ACCESS_DESCRIPTION_value(info, i);
          if (OBJ_obj2nid(ad->method) == NID_ad_ca_issuers && ad->location->type == GEN_URI) {
              ASN1_IA5STRING *uri = ad->location->d.uniformResourceIdentifier;
 diff --git a/test/unit/fapi-eventlog.c b/test/unit/fapi-eventlog.c
-index 3b859e39..796894ee 100644
+index 7e49e112..d3077a96 100644
 --- a/test/unit/fapi-eventlog.c
 +++ b/test/unit/fapi-eventlog.c
-@@ -130,6 +130,7 @@ check_eventlog_pcr0(const char *file, uint32_t *pcr_list, size_t pcr_list_size,
-     uint8_t *eventlog;
-     size_t size;
-     json_object *json_event_list = NULL;
-+#if HAVE_EVP_SM3
-     IFAPI_PCR_REG pcrs[TPM2_MAX_PCRS];
- 
-     TPML_PCR_SELECTION pcr_selection =
-@@ -149,19 +150,20 @@ check_eventlog_pcr0(const char *file, uint32_t *pcr_list, size_t pcr_list_size,
-          .buffer = { 0x15, 0xf4, 0xe6, 0xca, 0x45, 0x7d, 0x1a, 0xf6, 0xbc, 0x49,
-                      0x51, 0x1a, 0x93, 0xba, 0x35, 0x00, 0xad, 0x69, 0xac, 0xc5 },
-         };
--
-+#endif
-     /* Read file to get file size for comparison. */
-     eventlog = file_to_buffer(file, &size);
-     assert_non_null(eventlog);
- 
-     r = ifapi_get_tcg_firmware_event_list(file, pcr_list, pcr_list_size, &json_event_list);
-     assert_int_equal (r, TSS2_RC_SUCCESS);
--
-+#if HAVE_EVP_SM3
-     r = ifapi_calculate_pcrs(json_event_list, &pcr_selection, TPM2_ALG_SHA1, NULL, &pcrs[0]);
-     assert_int_equal (r, TSS2_RC_SUCCESS);
- 
-     /* Compare with the pcr0 value got from system with HCRTM events */
-     assert_true(!memcmp(&expected_pcr0.buffer[0], &pcrs[0].value.buffer[0], 20));
-+#endif
- 
-     json_object_put(json_event_list);
-     SAFE_FREE(eventlog);
-@@ -171,7 +173,7 @@ static void
- check_bios_hcrtm(void **state)
- {
+@@ -163,7 +163,7 @@ check_eventlog_pcr0(const char *file, uint32_t *pcr_list, size_t pcr_list_size,
+ static void
+ check_bios_hcrtm(void **state) {
  
 -#ifdef __FreeBSD__
 +#ifndef HAVE_EVP_SM3
```

Comment:
Is this still needed? I don't see `#if HAVE_EVP_SM3` added to the patch as before.

Reply:
Yes. AWS-LC does not have SM3, so we need the `skip()` on line 95 used.